### DISCUSSION TITLE
Should there be a warning for improperly formatted CAS numbers and/or should they be automatically reformatted?

### DISCUSSION THREAD

**Discussion by ethanbass** [ropensci/webchem#396] [General]
It seems that most `webchem` functions do not currently check the formatting of CAS numbers. For example,  `cts_convert("1333740", from='CAS', to="ChemSpider")` returns NA, but if you reformat the CAS number `cts_convert(webchem::as.cas("1333740"), from='CAS', to="ChemSpider")` the function returns several chemspider IDs. However, some functions (like `nist_ri`) internally reformat the CAS number by running it through the `as.cas` function.

While CAS numbers are normally written with hyphens, it is also fairly common to see CAS numbers "in the wild" written as one long string of numbers without hyphens. I am thinking it would be helpful to throw a warning for improperly formatted CAS numbers and/or automatically run them through `as.cas` to ensure they are formatted correctly, rather than returning NA for CAS numbers that are "improperly" formatted.

**Comment by stitam** [comment]
Thanks @ethanbass, I like the idea to automatically format everywhere, the impact would be less characters of code to reach the end result and less frustration. Maybe `as.cas()` should throw a message (if verbose) if it cannot convert a string to CAS? Wondering if there is anything against automatic formatting?

**Reply by Aariq** [reply]
`is.cas()` already provides messages giving details of why something isn't a CAS number.  `as.cas()` already calls `is.cas()`—it would just need a `verbose` argument added to pass down to `is.cas()` probably, or `vebose=TRUE` could be hard-coded inside of `as.cas()`

e.g.:

``` r
library(webchem)
is.cas("12-3-12", verbose = TRUE)
#> 12-3-12: Second part should have two digits!
#> 12-3-12 
#>   FALSE
```

<sup>Created on 2023-05-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by ethanbass** [reply]
I took a crack at this. Working on a pull request

**Reply by ethanbass** [reply]
https://github.com/ropensci/webchem/pull/398
